actors/builtin/market/market_test.go

```diff
@@ -1359,6 +1389,31 @@ func TestCronTickDealExpiry(t *testing.T) {
 	endEpoch := abi.ChainEpoch(300)
 	sectorExpiry := abi.ChainEpoch(400)
 
+	t.Run("deal expiry -> deal is correctly processed twice in the same crontick", func(t *testing.T) {
+		end := abi.ChainEpoch(151) // 50 + 101
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		dealId := actor.publishAndActivateDeal(rt, client, mAddrs, startEpoch, end, 0, sectorExpiry)
+		d := actor.getDealProposal(rt, dealId)
+
+		// move the current epoch to startEpoch so next cron epoch will be start + 100 = 150
+		current := startEpoch
+		rt.SetEpoch(current)
+		pay, slashed := actor.cronTickAndAssertBalances(rt, client, provider, current, dealId)
+		require.EqualValues(t, big.Zero(), pay)
+		require.EqualValues(t, big.Zero(), slashed)
```

Comment:
Should be a check after this cron tick to ensure the deal exists

Reply:
This is done.